### DISCUSSION TITLE
Delay the rearrangement of habits when status is updated and sort order is by status

### DISCUSSION THREAD

**Discussion by navinthomas** [iSoron/uhabits#1357] [Feature Requests] [answered]
### Context
**Enable skip days:** Enabled
**Show question marks for missing data:** Enabled

When habits are sorted by status, and a user wants to mark a habit as lapsed or skipped, it can be quite difficult when done from the main home UI. To mark a habit as skipped, it needs two presses - the first press will mark it as done, and then the second one will mark it as skipped. Since the habits are sorted by status, the unmarked habits with a question mark are at the top initially. When a user wants to mark a habit as skipped, the user toggles the status of the corresponding habit, which is unmarked and at the top of the list at this point. Since toggling changes its status to done, the app immediately sorts it and moves it down to the other habits with done status. Now the user has to scroll down and find the habit, since they have to toggle the status once more to mark it as skipped. This need for scrolling downwards and then searching for the concerned habit can be quite confusing when there are a lot of habits. This inconvenience becomes worse when the user wants to mark a habit as lapsed. That requires three toggles. So, the habit keeps getting shuffled around while the user tries to toggle the status from done -> skipped -> lapsed.

### Suggestion
When the sort order is by status, and a habit's status is updated, add a short delay before the habit is sorted/moved in to it's correct position. This short delay would give enough time for a user to change the status to what they want, before the sorting kicks in and moves the habit to it's correct final place.

**Comment by hiqua** [comment, marked as answer]
Didn't read everything, but I think the problem is addressed by the prompt shown e.g. in https://github.com/iSoron/uhabits/pull/1346 isn't it?

**Reply by navinthomas** [reply]
Yes, such a prompt should address this problem.
